### PR TITLE
Transformations: Add context parameter to transformDataFrame and operators

### PR DESCRIPTION
**Pull request by torkelo**
This community PR https://github.com/grafana/grafana/pull/44486, merged earlier this year, added variable interpolation capability to transformations but did it in a pretty hacky way, by passing the interpolation function in each transformation config object, which transformDataFrame then extracted and passed as an optional argument to each operator. 

I am exploring adding more context args to transformations (queries), to make it possible for transformations to access the queries.  And feel it is a good opportunity to fix this hacky way of passing the interpolation function. 

This is a minor breaking change if plugins use transformations directly (Which they should not, should be used via transformDataFrame), and if they use transformDataFrame it should not be a breaking change.

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
LGTM

**Comment by chriswue** [comment]
The ability to allow variable interpolation in transformers is currently a bit spread across multiple issues and discussions like #46428 and #52219
There has also been a PR #51804 that does it the same hacky way if I understand it correctly.

Does this PR (#60694) basically lay the ground work?

**Review by ryantxu — APPROVED** [review]
LGTM

File: packages/grafana-data/src/types/transformations.ts [hunk 1]
```diff
@@ -2,8 +2,17 @@ import { MonoTypeOperatorFunction } from 'rxjs';
...
 import { DataFrame, Field } from './dataFrame';
+import { InterpolateFunction } from './panel';
+import { DataQuery } from './query';
+
+/**
+ * Context passed to transformDataFrame and to each transform operator
+ */
+export interface DataTransformContext {
+  interpolate: InterpolateFunction;
+  queries?: DataQuery[];
```
Comment (ryantxu) [inline packages/grafana-data/src/types/transformations.ts:14]: ^^^ should the queries be removed/commented out since it is not yet used or implemented?

Comment (torkelo) [reply packages/grafana-data/src/types/transformations.ts:14]: yes

**Comment by torkelo** [comment]
> Does this PR (https://github.com/grafana/grafana/pull/60694) basically lay the ground work to do it correctly?

yes

**Pull request merged**